nuitka/nodes/IterationHandles.py

```diff
@@ -0,0 +1,310 @@
+        self.iter = iter(self.constant)
+
+    def __repr__(self):
+        return "<%s of %r>" % (self.__class__.__name__, self.constant_node)
+
+    def getNextValueExpression(self):
+        """Return the next iteration value or StopIteration exception
+        if the iteration has reached the end
+        """
+        try:
+            from .ConstantRefNodes import makeConstantRefNode
+
+            return makeConstantRefNode(
+                constant=next(self.iter), source_ref=self.constant_node.source_ref
+            )
+        except StopIteration:
+            return None
+
+    def getNextValueTruth(self):
+        """Return the boolean value of the next iteration handle."""
+        try:
+            iteration_value = next(self.iter)
+        except StopIteration:
+            return StopIteration
+        return bool(iteration_value)
+
+    def getIterationValueWithIndex(self, value_index):
+        return None
+
+
+class ConstantIndexableIterationHandle(ConstantIterationHandleBase):
+    """Class for the constants that are indexable.
+
+    Attributes
+    ----------
+    constant_node : node_object
+        Instance of the calling node.
+
+    Methods
+    -------
+    getIterationValueWithIndex(value_index)
+        Sequential access of the constants
+    """
+
+    def __init__(self, constant_node):
+        ConstantIterationHandleBase.__init__(self, constant_node)
+        assert type(self.constant) not in (set, dict)
+
+    def getIterationValueWithIndex(self, value_index):
+        """Tries to return constant value at the given index.
+
+        Parameters
+        ----------
+        value_index : int
+            Index value of the element to be returned
+        """
+        try:
+            from .ConstantRefNodes import makeConstantRefNode
+
+            return makeConstantRefNode(
+                constant=self.constant[value_index],
+                source_ref=self.constant_node.source_ref,
+            )
+        except IndexError:
+            return None
+
+
+class ConstantSetAndDictIterationHandle(ConstantIterationHandleBase):
+    """Class for the set and dictionary constants.
+    """
+
+    def __init__(self, constant_node):
+        ConstantIterationHandleBase.__init__(self, constant_node)
+        assert type(self.constant) in (set, dict)
+
+
+class ListAndTupleContainerMakingIterationHandle(IterationHandleBase):
+    """Class for list and tuple container making expression
+
+    Attributes
+    ----------
+    constant_node : node_object
+        Instance of the calling node.
+
+    Methods
+    -------
+    __repr__()
+        Prints representation of the ListAndTupleContainerMakingIterationHandle
+        object
+    getNextValueExpression()
+        Returns the next iteration value
+    getNextValueTruth()
+        Returns the boolean value of the next handle
+    getIterationValueWithIndex(value_index)
+        Sequential access of the expression
+    """
+
+    def __init__(self, constant_node):
```

Comment:
So this one is getting a list with the element values, instead of a node, right? So why is it named `constant_node`, that seems misleading.